llama-index-integrations/indices/llama-index-indices-managed-vectara/llama_index/indices/managed/vectara/query.py

```diff
@@ -129,3 +149,98 @@ def _get_prompt_modules(self) -> PromptMixinType:
 
     def _update_prompts(self, prompts: PromptDictType) -> None:
         """Update prompts."""
+
+
+class VectaraChatEngine(BaseChatEngine):
+    def __init__(
+        self,
+        retriever: VectaraRetriever,
+        streaming: bool = False,
+        summary_response_lang: str = "eng",
+        summary_num_results: int = 5,
+        summary_prompt_name: str = "vectara-summary-ext-24-05-sml",
+        node_postprocessors: Optional[List[BaseNodePostprocessor]] = None,
+        callback_manager: Optional[CallbackManager] = None,
+        **kwargs: Any,
+    ) -> None:
+        self._retriever = retriever
+        self._streaming = streaming
+        self._summary_enabled = True
+        self._summary_response_lang = summary_response_lang
+        self._summary_num_results = summary_num_results
+        self._summary_prompt_name = summary_prompt_name
+        self._node_postprocessors = node_postprocessors or []
+
+        self.callback_manager = callback_manager or CallbackManager([])
+        for node_postprocessor in self._node_postprocessors:
+            node_postprocessor.callback_manager = self.callback_manager
+
+        self.conv_id = None
+
+    @classmethod
+    def from_args(
+        cls,
+        retriever: VectaraRetriever,
+        streaming: bool = False,
+        service_context: Optional[ServiceContext] = None,
+        node_postprocessors: Optional[List[BaseNodePostprocessor]] = None,
+        **kwargs: Any,
+    ) -> "VectaraChatEngine":
+        """Initialize a ContextChatEngine from default parameters."""
+        node_postprocessors = node_postprocessors or []
+        return cls(
+            retriever,
+            streaming,
+            node_postprocessors=node_postprocessors,
+            callback_manager=callback_manager_from_settings_or_context(
+                Settings, service_context
+            ),
+            **kwargs,
+        )
+
+    def chat(self, message: str) -> AgentChatResponse:
+        """Chat with the agent."""
+        with self.callback_manager.event(
+            CBEventType.QUERY, payload={EventPayload.QUERY_STR: message}
+        ) as query_event:
+            kwargs = (
+                {
+                    "summary_response_lang": self._summary_response_lang,
+                    "summary_num_results": self._summary_num_results,
+                    "summary_prompt_name": self._summary_prompt_name,
+                }
+                if self._summary_enabled
+                else {}
+            )
+            nodes, summary, self.conv_id = self._retriever._vectara_query(
+                QueryBundle(message), chat=True, conv_id=self.conv_id, **kwargs
```

Comment:
vector is managing the memory here I'm guessing?

Reply:
Yes Vectara is doing all the history management and rephrasing behind the scenes. Pretty cool, no?